tests/main/hidden-snap-dir/task.yaml

```diff
@@ -8,13 +8,79 @@ environment:
     NAME: test-snapd-tools
 
 prepare: |
-    snap pack "$TESTSLIB"/snaps/"$NAME"
+    snap pack "$TESTSLIB/snaps/$NAME"
     "$TESTSTOOLS"/snaps-state install-local "$NAME"
 
 restore: |
     snap unset system experimental.hidden-snap-folder
 
 execute: |
+    # core22 snap isn't available for x86
+    if os.query is-pc-i386; then
+      exit 0
+    fi
+
+    # Checks that the env vars are as expected after the migration. 1st parameter
+    # is the current revision; 2nd parameter toggles ~/Snap related checks
+    # ("true" does the checks, anything else doesn't)
+    check_env() {
+      echo "Check the env vars were migrated"
+
+      snapEnv=$("$NAME".env)
+      echo "$snapEnv" | MATCH "SNAP_USER_DATA=$HOME/\.snap/data/$NAME/$1"
+      echo "$snapEnv" | MATCH "SNAP_USER_COMMON=$HOME/\.snap/data/$NAME/common"
+
+      if [ "$2" = "true" ]; then
```

Comment:
I would use `"yes"` instead of `"true"`; not that it really matters, but `true` is a bit overloaded, and especially when booleans are written in quotes, someone who's not familiar with shell scripting might think that `"false"` evaluates to `true` (like it would do in C++ if one wrote `bool val = "false";` -- yes, once I had a colleague who did that).

At least, that's the more common style for bash scripts.